### ISSUE TITLE
Create file 8TB that use entire disk space fails

### ISSUE THREAD

**Issue by xsoft** [veracrypt/VeraCrypt#878]
VeraCrypt don't use entire disk space when asked to.
Also created (quoter-size file) is damaged, without ability to mount.

# Expected behavior

Create 8TB file (encrypted container), ext3,no hidden volume, that use entire disk space, and its mountable at the end (not corrupted). Sounds easy enough. But fails on 3rd attempt, when using GUI on linux, 9-12h each time.

---

When creating a file container on Linux, with option to use entire free disk space, it still create just 2TB file max, on 8TB drive.
When this creation takes like 10h to make, it is really unpleasant to repeate and waste another day.
And using a bite-like method can be interrupted at the end, because one byte is missing (GB/GiB...)

Also files created like this (which are interrupted in the middle), and with mounting errors like:
```Error: device-mapper: reload ioctl on veracrypt2 (254:0) failed: Invalid argument```
or
```cant mount ext3 wrong fs type, bad option, bad superblock```

Note that when creating just 1GB file, with same steps, all works fine.

Also that mounting is done at same way like other disks/files created before.
```
veracrypt --password="..." --pim=0 -k "" --protect-hidden=no /media/root/kali2-hdd3/file.txt /media/veracrypt2
(throws errors)

veracrypt --password="..." --pim=0 -k "" --protect-hidden=no /media/root/a163...b19/file.txt /media/veracrypt1
(previous drive, works fine)
```

File is created on clean 8TB EXT4 drive.

```
root@hp-kali2:/media# fdisk -l
Disk /dev/sdb: 7.28 TiB, 8001563222016 bytes, 15628053168 sectors
Disk model: ST8000xxxxx
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 4096 bytes
I/O size (minimum/optimal): 4096 bytes / 4096 bytes
^---this drive--with errors---

Disk /dev/sda: 931.51 GiB, 1000204886016 bytes, 1953525168 sectors
Disk model: SAMSUNG HDxxxxx
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
^---another-drive---success -mouting---
```

# Observed behavior

File is created. Message is with big red "!" symbol, but it says that operation is finished.
But result file is .. .damaged, unussable. Also with wrong size.

And no "expand file" or "fix file" is available.
Google throws that extend file feature was somewhere at 2015, but none Linux solution until now.
And yeah, "create another file and copy" is really great. Especially when Im trying to do that now. And migrate 1TB file to 8TB file. Fist, good luck with finding 8TB spare drive. Second, good luck with 3 days going back and forward to migrate those huge amount of data. 8GB would be cool and fast to do, 8TB is really a paint. And no support on reliable tools are available. Not even "use entire disk space" checkbox cant be trusted. Sorry for be angry a bit, but this is 3rd night, wasted.

# Your Environment

**VeraCrypt version:** VeraCrypt 1.25.4
**Operating system and version:** Kali Linux 2021
**System type:** Linux hp-kali2 5.15.0-kali2-amd64 #1 SMP Debian 5.15.5-2kali2 (2021-12-22) x86_64 GNU/Linux

**Comment by xsoft** [comment]
Nope. 4th fail when using commandline this time.

```veracrypt  --text --create /media/root/x1/test.txt --size 7360646324000 --volume-type=normal --encryption=aes --hash=sha-512 --filesystem=ext3  --pim=0 --keyfiles "" --random-source=/dev/urandom

-rw-------  1 root root 2194719883264 Jan 21 23:51  test.txt

Filesystem      Size  Used Avail Use% Mounted on
/dev/sdb        7.3T  2.0T  4.9T  30% /media/x1/kali2-hdd3```

tldr: 2TB file is created, when 7,3TB file is requested.

**Comment by JonasTrampe** [comment]
The underlying filesystem does not allow for files to be larger than 2 TiB.

**Comment by idrassi** [comment]
@xsoft 

This seems to be a limitation of the filesystem. To be sure, can you please run the following commands on the root on the 8TB drive:
```
fallocate -l 7T large_7TB.img
mkfs.ext3 large_7TB.img
```
the fallocate command is fast because it doesn't write data to disk.

Does it work?

**Comment by idrassi** [comment]
Closing for now. Feel free to reopen if you have other elements indicating that it is a VeraCrypt issue.
